progressReport/app.py

```diff
@@ -126,17 +161,24 @@ def assign_node_levels(node, student_levels_count, class_levels_count):
                            course_data=course_nodes)
 
 
-@app.route('/parse', methods=["POST"])
-def parse():
-    school_name = request.args.get("school_name", "Berkeley")
-    course_name = request.form.get("course_name", "CS10")
-    parser.generate_map(school_name=secure_filename(school_name), course_name=secure_filename(course_name), render=False)
-    try:
-        with open("data/{}_{}.json".format(secure_filename(school_name), secure_filename(course_name))) as data_file:
-            course_data = json.load(data_file)
-    except FileNotFoundError:
-        return "Class not found", 400
-    return course_data
+@app.route('/', methods=["GET"])
+def index():
+    school_name = request.args.get("school", "Berkeley")
+    course_name = request.args.get("class", "CS10")
+    student_mastery = request.args.get("student_mastery", "0")
+    class_mastery = request.args.get("class_mastery", "0")
+    return render(school_name, course_name, student_mastery, class_mastery)
+
+@app.route('/json', methods=["POST"])
+def index_json():
+    args_json = request.get_json()
+    if args_json is None:
+        return "URL JSON parameter is invalid", 400
+    school_name = args_json["school"] if "school" in args_json else "Berkeley"
+    course_name = args_json["class"] if "class" in args_json else "CS10"
+    student_mastery = args_json["student_mastery"] if "student_mastery" in args_json else {}
+    class_mastery = args_json["class_mastery"] if "class_mastery" in args_json else {}
+    return render(school_name, course_name, student_mastery, class_mastery, use_json=True)
```

Comment:
The data should be submitted as a post request, but `POST` should not render HTML (even if it is possible).  This is a general rule of thumb and I mentioned it to you in person, but it doesn't seem to have been documented in the ticket directly.  [See here](https://en.wikipedia.org/wiki/Post/Redirect/Get) for more information about this design pattern.

Reply:
resolved